FRAuth/FRAuth/DeviceBinding/DeviceBindingAuthenticators.swift

```diff
@@ -0,0 +1,250 @@
+        var keyBuilderQuery = keyAware.keyBuilderQuery()
+        keyBuilderQuery[String(kSecAttrAccessControl)] = accessControl()
+        
+        let context = LAContext()
+        context.localizedReason = promptDescription
+        keyBuilderQuery[String(kSecUseAuthenticationContext)] = context
+        
+        return try keyAware.createKeyPair(builderQuery: keyBuilderQuery)
+    }
+    
+    
+    /// Check if authentication is supported
+    func isSupported() -> Bool {
+        let laContext = LAContext()
+        var evalError: NSError?
+        return laContext.canEvaluatePolicy(policy, error: &evalError)
+    }
+    
+    
+    /// Access Control for the authetication type
+    func accessControl() -> SecAccessControl? {
+#if !targetEnvironment(simulator)
+        return SecAccessControlCreateWithFlags(kCFAllocatorDefault, kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly, SecuredKey.isAvailable() ? [.biometryCurrentSet, .privateKeyUsage] : [.biometryCurrentSet], nil)!
+#else
+        return nil
+#endif
+    }
+}
+
+
+/// DeviceAuthenticator adoption for biometric and Device Credential authentication
+internal struct BiometricAndDeviceCredential: DeviceAuthenticator {
+    /// prompt description for authentication promp if applicable
+    var promptDescription: String
+    /// local authentication policy for authentication
+    var policy: LAPolicy
+    /// keyAware for key pair generation
+    var keyAware: KeyAware
+    
+    
+    /// Initializes BiometricOnly with given BiometricHandler and KeyAware
+    /// - Parameter biometricInterface: biometric handler for authentication
+    /// - Parameter keyAware: keyAware for key pair generation
+    init(description: String, keyAware: KeyAware) {
+        self.promptDescription = description
+        self.keyAware = keyAware
+        policy = .deviceOwnerAuthentication
+    }
+    
+    
+    /// Generate public and private key pair
+    func generateKeys() throws -> KeyPair {
+        var keyBuilderQuery = keyAware.keyBuilderQuery()
+        keyBuilderQuery[String(kSecAttrAccessControl)] = accessControl()
+        
+        let context = LAContext()
+        context.localizedReason = promptDescription
+        keyBuilderQuery[String(kSecUseAuthenticationContext)] = context
+        
+        return try keyAware.createKeyPair(builderQuery: keyBuilderQuery)
+    }
+    
+    
+    /// Check if authentication is supported
+    func isSupported() -> Bool {
+        let laContext = LAContext()
+        var evalError: NSError?
+        return laContext.canEvaluatePolicy(policy, error: &evalError)
+    }
+    
+    
+    /// Access Control for the authetication type
+    func accessControl() -> SecAccessControl? {
+#if !targetEnvironment(simulator)
+        return SecAccessControlCreateWithFlags(kCFAllocatorDefault, kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly, SecuredKey.isAvailable() ? [.userPresence, .privateKeyUsage] : [.userPresence], nil)!
+#else
```

Comment:
i am thinking like if its debug mode use this  [.userPresence] and for release [.userPresence, .privateKeyUsage] ...so we can remove the SecureEnclave check ...thoughts ?